test/jasmine/tests/ternary_test.js

```diff
@@ -382,6 +382,76 @@ describe('ternary plots', function() {
         .then(done);
     });
 
+    it('should be able to hide/show ticks and tick labels', function(done) {
+        var gd = createGraphDiv();
+        var fig = Lib.extendDeep({}, require('@mocks/ternary_simple.json'));
+
+        function assertCnt(selector, expected, msg) {
+            var sel = d3.select(gd).selectAll(selector);
+            expect(sel.size()).toBe(expected, msg);
+        }
+
+        function toggle(selector, astr, vals, exps) {
+            return Plotly.relayout(gd, astr, vals[0]).then(function() {
+                assertCnt(selector, exps[0], astr + ' ' + vals[0]);
+                return Plotly.relayout(gd, astr, vals[1]);
+            })
+            .then(function() {
+                assertCnt(selector, exps[1], astr + ' ' + vals[1]);
+                return Plotly.relayout(gd, astr, vals[0]);
+            })
+            .then(function() {
+                assertCnt(selector, exps[0], astr + ' ' + vals[0]);
+            });
+        }
+
+        Plotly.plot(gd, fig)
+        .then(function() {
+            return toggle(
+                '.aaxis > .ytick > text',
+                'ternary.aaxis.showticklabels',
+                [true, false], [4, 0]
+            );
```

Comment:
Minor style comment: you can put the extra function wrapping inside the definition of `toggle`, so that the individual calls are just `.then(toggle(...))`

Regardless `toggle` is a very nice simplification!

Reply:
Good call. Done in -> 68e0628b20e2bf8f05eaac28e094e60584669f2b